core/src/test/java/org/sonarsource/sonarlint/core/container/connected/update/ProjectBranchesDownloaderTest.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * SonarLint Core - Implementation
+ * Copyright (C) 2016-2021 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonarsource.sonarlint.core.container.connected.update;
+
+import java.util.Collection;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+import org.sonarsource.sonarlint.core.MockWebServerExtension;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+class ProjectBranchesDownloaderTest {
+
+  @RegisterExtension
+  static MockWebServerExtension mockServer = new MockWebServerExtension();
+
+  private final static String PROJECT_KEY = "project1";
+
+  private ProjectBranchesDownloader underTest;
+
+  @BeforeEach
+  public void setUp() {
+    underTest = new ProjectBranchesDownloader(mockServer.serverApiHelper());
+  }
+
+  @Test
+  void shouldDownloadBranches() {
+    mockServer.addStringResponse("/api/project_branches/list?project=project1",
+      "{\n" +
+        "  \"branches\": [\n" +
+        "    {\n" +
+        "      \"name\": \"feature/foo\",\n" +
+        "      \"isMain\": false,\n" +
+        "      \"type\": \"BRANCH\",\n" +
+        "      \"status\": {\n" +
+        "        \"qualityGateStatus\": \"OK\"\n" +
+        "      },\n" +
+        "      \"analysisDate\": \"2017-04-03T13:37:00+0100\",\n" +
+        "      \"excludedFromPurge\": false\n" +
+        "    },\n" +
+        "    {\n" +
+        "      \"name\": \"master\",\n" +
+        "      \"isMain\": true,\n" +
```

Comment:
Note: I wonder if we'll need this attribute, e.g to decide what the "fallback" should be when we cannot find the appropriate branch.

Reply:
Yep. Me too. I decided not to extract all data we will possibly need to avoid redundant code and get only the name for now. We can extract more when we will know exactly what we need.